src/avram/nested_save_operation.cr

```diff
@@ -1,35 +1,61 @@
 module Avram::NestedSaveOperation
   macro has_one(type_declaration)
     {% name = type_declaration.var %}
-    {% type = type_declaration.type %}
-    @_{{ name }} : {{ type }} | Nil
+    {% type = type_declaration.type.resolve %}
+    {% model_type = type.superclass.superclass.type_vars.first.resolve %}
 
-    def {{ name }}
-      @_{{ name }} ||= {{ type }}.new(params)
-    end
+    {% assoc = T.resolve.constant(:ASSOCIATIONS).find do |assoc|
+         assoc[:relationship_type] == :has_one &&
+           assoc[:type].resolve.name == model_type.name
+       end %}
```

Comment:
I'm worried about this section. I'm fairly certain if you call `.resolve` on a class from a file that hasn't been required yet it will fail. That in itself is not terrible it's just something that we've/I've tried to avoid.

That being said, calling `.superclass.superclass` is understandable but I think we want to avoid it just in case someone subclasses a subsclass of an operation.  What I **think** you can do is call `type.ancestors[-3]` instead

```crystal
class Foo
end

class Bar < Foo
end

{{ Bar.ancestors }} # [Foo, Reference, Object]
{{ Bar.ancestors[-3] }} # Foo
```
Not entirely sure but I think that's what you want.

I just wish this could be solved without digging so far into all these classes 😬

Reply:
> What I think you can do is call `type.ancestors[-3]` instead

I like this better. I'll give it a shot.

> I'm fairly certain if you call `.resolve` on a class from a file that hasn't been required yet it will fail.

I'm not sure what else is expected. I'll try doing away with `.resolve` and compare strings instead. We'll see.

Reply:
Doesn't work.

This:

```crystal
{% puts type.ancestors %}
```
...produces:
```bash
[EmailAddress::SaveOperation, Avram::SaveOperation(EmailAddress), Avram::InheritColumnAttributes, Avram::MarkAsFailed, Avram::NestedSaveOperation, Avram::DatabaseValidations(EmailAddress), Avram::Callbacks, Avram::NeedyInitializerAndSaveMethods, Avram::ParamKeyOverride, Avram::OperationErrors, Avram::Validations, Avram::DefineAttribute, Reference, Object, Spec::ObjectExtensions, Colorize::ObjectExtensions]
```

It seems all included modules become part of the type hierarchy in a way that makes it unreliable.

Reply:
I got it. This works:

```crystal
{% model_type = type.ancestors.find do |t|
  t.stringify.starts_with?("Avram::SaveOperation(")
end.type_vars.first.resolve %}
```